### PR TITLE
Improve job config error when no handler elements are defined.

### PR DESCRIPTION
**Pull request by jmchilton**
NoneValue has not attribute 'get' isn't super helpful in tracking that problem down.

### CODE REVIEW

**Comment by bgruening** [comment]
:+1:

File: lib/galaxy/jobs/__init__.py [hunk 1]
```diff
@@ -192,6 +192,9 @@ def __parse_job_conf_xml(self, tree):
                             else:
                                 self.handlers[tag] = [id]
 
+        # Must define at least one handler to have a default.
+        if not self.handlers:
+            raise ValueError("Job configuration file defines no valid handler elements.")
```
Comment (nsoranzo) [inline lib/galaxy/jobs/__init__.py:197]: Maybe `ConfigurationError` from `galaxy.exceptions` would be a better exception to raise?

Comment (jmchilton) [inline lib/galaxy/jobs/__init__.py:197]: My thought was this was very low-level code we may want to rip out and reuse other places - the planemo of job configurations for instance. For this reason, I didn't want to create that inter-dependency unnecessarily. Also this exception will never be thrown from the API so the advantages of galaxy.exceptions are minimized. That said `ValueError` isn't quite right either - any compromises in the Python stdlib or do you feel strongly that we should use galaxy.exceptions?

Comment (nsoranzo) [inline lib/galaxy/jobs/__init__.py:197]: Looking better at the whole file, I think `ValueError` is fine here, it's the `config_exception()` function that should raise `ConfigurationError` at https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/__init__.py#L101 . Is that also something you plan to use in planemo?

**Pull request merged**